optaplanner-core/src/test/java/org/optaplanner/core/config/solver/SolverConfigMultiThreadedTest.java

```diff
@@ -0,0 +1,121 @@
+package org.optaplanner.core.config.solver;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.config.solver.testutil.TestThreadFactory;
+import org.optaplanner.core.impl.testdata.domain.TestdataEntity;
+import org.optaplanner.core.impl.testdata.domain.TestdataSolution;
+import org.optaplanner.core.impl.testdata.domain.TestdataValue;
+import org.optaplanner.core.impl.testdata.util.PlannerTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class SolverConfigMultiThreadedTest {
+
+    @Test
+    public void moveThreadCountAutoIsPositive() {
+        final int cpuCount = 16;
+        assertThat(mockSolverConfigForMoveThreadCountAuto(cpuCount).resolveMoveThreadCount()).isEqualTo(cpuCount - 2);
+    }
+
+    @Test
+    public void moveThreadCountAutoIsNegative() {
+        final int cpuCount = -2;
+        assertThat(mockSolverConfigForMoveThreadCountAuto(cpuCount).resolveMoveThreadCount()).isNull();
```

Comment:
I can't help it, the test name is misleading. `move thread count auto` IS `negative`??? Actually `resolved move thread count` IS `null` WHEN `the resolution strategy` IS `auto` AND `available cpu count` IS `negative`.

Reply:
@yurloc although being much longer, this variant is much more descriptive. Updated the naming according to your suggestion. Thanks!